test/common/validation.go

```diff
@@ -0,0 +1,119 @@
+)
+
+const (
+	// The interval and timeout used for polling pod logs.
+	interval = 1 * time.Second
+	timeout  = 2 * time.Minute
+)
+
+// CheckLogContent waits until logs for the logger Pod include the given content.
+// If the content is not present within timeout it returns error.
+func (client *Client) CheckLogContent(podName, content string) error {
+	namespace := client.Namespace
+	containerName, err := client.getContainerName(podName, namespace)
+	if err != nil {
+		return err
+	}
+	return wait.PollImmediate(interval, timeout, func() (bool, error) {
+		logs, err := client.Kube.PodLogs(podName, containerName, namespace)
+		if err != nil {
+			return true, err
+		}
+		return strings.Contains(string(logs), content), nil
+	})
+}
+
+// CheckLogContents waits until logs for the logger Pod include the given contents.
+// If the contents are not present within timeout it returns error.
+func (client *Client) CheckLogContents(podName string, contents []string) error {
+	namespace := client.Namespace
+	containerName, err := client.getContainerName(podName, namespace)
+	if err != nil {
+		return err
+	}
+	return wait.PollImmediate(interval, timeout, func() (bool, error) {
+		logs, err := client.Kube.PodLogs(podName, containerName, namespace)
+		if err != nil {
+			return true, err
+		}
+		for _, content := range contents {
+			if !strings.Contains(string(logs), content) {
+				return false, nil
+			}
+		}
+		return true, nil
+	})
+}
+
+// CheckLogContentCount checks if the number of substr occur times equals the given number.
+// If the content does not appear the given times it returns error.
+func (client *Client) CheckLogContentCount(podName, content string, appearTimes int) error {
+	namespace := client.Namespace
+	containerName, err := client.getContainerName(podName, namespace)
+	if err != nil {
+		return err
+	}
+	return wait.PollImmediate(interval, timeout, func() (bool, error) {
+		logs, err := client.Kube.PodLogs(podName, containerName, namespace)
+		if err != nil {
+			return true, err
+		}
+
+		return strings.Count(string(logs), content) == appearTimes, nil
+	})
+}
+
+// FindAnyLogContents attempts to find logs for given Pod/Container that has 'any' of the given contents.
+// It returns an error if it couldn't retrieve the logs. In case 'any' of the contents are there, it returns true.
+func (client *Client) FindAnyLogContents(podName string, contents []string) (bool, error) {
+	namespace := client.Namespace
+	containerName, err := client.getContainerName(podName, namespace)
+	if err != nil {
+		return false, err
+	}
+	logs, err := client.Kube.PodLogs(podName, containerName, namespace)
+	if err != nil {
+		return false, err
+	}
+	for _, content := range contents {
+		if strings.Contains(string(logs), content) {
+			client.Logf("Found content %q for %s/%s.", content, podName, containerName)
+			return true, nil
+		}
+	}
+	return false, nil
+}
+
+// getContainerName gets name of the first container of the given pod.
```

Comment:
Why does this return the first container? Are we assuming some sort of semantics to the container ordering?

Reply:
Right now the logger pod only contains one container, and we use it in all test cases to receive events and validate.

Reply:
Can you add a comment here with the explanation you provided here?